base/loading.jl

```diff
@@ -1231,7 +1233,7 @@ function insert_extension_triggers(env::String, pkg::PkgId)::Union{Nothing,Missi
                         return _insert_extension_triggers(pkg, extensions, weakdeps)
                     end
 
-                    d_weakdeps = Dict{String, String}()
+                    d_weakdeps = Dict{String, Any}()
                     for (dep_name, entries) in d
                         dep_name in weakdeps || continue
```

Comment:
Does this line need to check if it is in the keys of weakdeps?

Reply:
`weakdeps` here is `Vector{String}` always since we handle the `weakdeps::Dict{String,Any}` case above.